RiotSwiftUI/Modules/Spaces/SpaceSettings/SpaceSettings/View/SpaceSettings.swift

```diff
@@ -0,0 +1,204 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+@available(iOS 14.0, *)
+struct SpaceSettings: View {
+
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    @Environment(\.theme) private var theme: ThemeSwiftUI
+    
+    // MARK: Public
+    
+    @ObservedObject var viewModel: SpaceSettingsViewModel.Context
+    
+    var body: some View {
+        ScrollView {
+            VStack {
+                avatarView
+                Spacer().frame(height:32)
+                formView
+                roomAccess
+                options
+            }
+        }
+        .background(theme.colors.navigation)
+        .waitOverlay(show: viewModel.viewState.isLoading, allowUserInteraction: false)
+        .ignoresSafeArea(.container, edges: .bottom)
+        .frame(maxHeight: .infinity)
+        .navigationBarBackButtonHidden(true)
+        .navigationTitle(VectorL10n.settingsTitle)
+        .toolbar {
+            ToolbarItem(placement: .confirmationAction) {
+                Button(VectorL10n.done) {
+                    updateSpace()
+                }
+                .disabled(!viewModel.viewState.isModified || !viewModel.viewState.isAddressValid)
+            }
+            ToolbarItem(placement: .cancellationAction) {
+                Button(VectorL10n.cancel) {
+                    viewModel.send(viewAction: .cancel)
+                }
+            }
+        }
+        .alert(isPresented: $viewModel.showPostProcessAlert, content: {
+            Alert(title: Text(VectorL10n.settingsTitle),
+                  message: Text(VectorL10n.spaceSettingsUpdateFailedMessage),
+                  primaryButton: .default(Text(VectorL10n.retry), action: {
+                    updateSpace()
+                  }),
+                  secondaryButton: .cancel())
+        })
+    }
+    
+    // MARK: - Private
+    
+    @ViewBuilder
+    private var avatarView: some View {
+        ZStack(alignment: .bottomTrailing) {
+            GeometryReader { reader in
+                ZStack {
+                    SpaceAvatarImage(mxContentUri: viewModel.viewState.avatar.mxContentUri, matrixItemId: viewModel.viewState.avatar.matrixItemId, displayName: viewModel.viewState.avatar.displayName, size: .xxLarge)
+                    .padding(6)
+                    if let image = viewModel.viewState.userSelectedAvatar {
+                        Image(uiImage: image)
+                            .resizable()
+                            .scaledToFill()
+                            .frame(width: 80, height: 80, alignment: .center)
+                            .clipShape(RoundedRectangle(cornerRadius: 8))
+                    }
+                }.padding(10)
+                .gesture(TapGesture().onEnded { _ in
```

Comment:
```suggestion
                .onTapGesture {
```

Or maybe as mentioned before, use a button with the `ZStack` as its label.